apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl

```diff
@@ -0,0 +1,34 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022-2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_resource_proto_v2).
+
+-behaviour(emqx_bpapi).
+
+-export([
+    introduced_in/0,
+    reset_metrics/1
+]).
```

Comment:
Q: shouldn't this have the other non-deprecated functions as well (i.e., `recreate` and `remove`)?

Reply:
I don't think so.
The older version nodes will call the `v1` module (which is not deleted yet) during rolling upgrade.

Reply:
ah, forgot to add comments.

Reply:
Deleted the "deprecated" comments.
We need the v1 module to stick around to pin the `_local` flavor APIs for xref and dialyzer checks.